server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/LoadReportAnalysisMetadataHolderStep.java

```diff
@@ -87,6 +103,43 @@ private void checkProjectKeyConsistency(ScannerReport.Metadata reportMetadata) {
     }
   }
 
+  private void checkOrganizationKeyConsistency(ScannerReport.Metadata reportMetadata, Organization organization) {
+    String organizationKey = reportMetadata.getOrganizationKey();
+    String resolveReportOrganizationKey = resolveReportOrganizationKey(organizationKey);
+    if (!resolveReportOrganizationKey.equals(organization.getKey())) {
+      if (reportBelongsToDefaultOrganization(organizationKey)) {
+        throw MessageException.of(format(
+          "Report does not specify an OrganizationKey but it has been submitted to another organization (%s) than the default one (%s)",
+          organization.getKey(),
+          defaultOrganizationProvider.get().getKey()));
+      } else {
+        throw MessageException.of(format(
+          "OrganizationKey in report (%s) is not consistent with organizationKey under which the report as been submitted (%s)",
+          resolveReportOrganizationKey,
+          organization.getKey()));
+      }
+    }
+  }
+
+  private String resolveReportOrganizationKey(@Nullable String organizationKey) {
+    if (reportBelongsToDefaultOrganization(organizationKey)) {
+      return defaultOrganizationProvider.get().getKey();
+    }
+    return organizationKey;
+  }
+
+  private static boolean reportBelongsToDefaultOrganization(@Nullable String organizationKey) {
+    return organizationKey == null || organizationKey.isEmpty();
+  }
+
+  private Organization toOrganization(String organizationUuid) {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      Optional<OrganizationDto> organizationDto = dbClient.organizationDao().selectByUuid(dbSession, organizationUuid);
+      checkState(organizationDto.isPresent(), "Organization with uuid '{}' can't be found", organizationUuid);
+      return Organization.from(organizationDto.get());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Call "organizationDto.isPresent()" before accessing the value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)